### DISCUSSION TITLE
Using Suggester for uncreated links/notes?

### DISCUSSION THREAD

**Discussion by SteveDockar** [chhoumann/quickadd#255] [Q&A]
Brief backstory: I often include the names of people in notes using [[First Last]]. I don't usually create the actual person/linked note but can still see the "person" in the Graph View with the "Existing files only" filter disabled. For people I deal with often, I like to modify the link to [[First Last|First]] so the note looks a little more personal and less formal, but if I do create them, the note file will be their full name for clarlity. When I do create a person file, I add "First" as an alias so this is more automated. 

I created a QuickAdd to capture "First Last" using inputPrompt and then use Javascript string wrangling to return "[[First Last|First]]", but would like to enhance it to allow me to pick from existing uncreated (people)notes like typing `[[<some-text` in the note editor does. I managed to get the example `suggester` API to show existing note/files in the vault, but couldn't figure out how to access Obsidian's resolved _and_ unresolved links (or anywhere else I can get suggestions from). I had a cursory look at the Obsidian API docs, but that's way deeper than my current level of knowledge.

Is something like this possible, and if so have I just failed to spot the helpful example or is there (much) more to it than that? 

Thanks in advance.

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

Yes, it's possible, and the piece you were missing is `app.metadataCache.unresolvedLinks`. It's a map of source file -> links that don't resolve to a real file yet, which is exactly where the editor's `[[` autocomplete gets uncreated notes from.

For your person-link flow, merge those with existing files and feed the result to the suggester. As an inline script (a ```js quickadd fence in your capture format):

````
```js quickadd
const names = new Set();
for (const file of this.app.vault.getMarkdownFiles()) {
    names.add(file.basename);
}
const unresolved = this.app.metadataCache.unresolvedLinks;
for (const source in unresolved) {
    for (const link in unresolved[source]) names.add(link);
}

const sorted = [...names].sort((a, b) => a.localeCompare(b));
const name = await this.quickAddApi.suggester(sorted, sorted, "Person", true);
if (!name) return "";
return `[[${name}|${name.split(" ")[0]}]]`;
```
````

In a user script it's the same code with `params.app` / `params.quickAddApi`. The `true` at the end enables custom input, so you can also type a brand-new name that isn't in the list yet.

Also worth knowing: QuickAdd's own prompts learned this trick since you asked. Typing `[[` in any QuickAdd input prompt opens a file search that includes unresolved links, just like the editor. See https://quickadd.obsidian.guide/docs/SuggesterSystem
